guidance/library/_json.py

```diff
@@ -115,18 +115,74 @@ def _process_additional_properties(
 def _gen_json_array(
     lm,
     *,
-    item_schema: Mapping[str, Any],
+    prefix_items_schema: Optional[Sequence[Mapping[str, Any]]],
+    item_schema: Optional[Mapping[str, Any]],
+    min_items: int,
+    max_items: Optional[int],
     definitions: Mapping[str, Callable[[], GrammarFunction]],
 ):
+    if prefix_items_schema is None:
+        prefix_items_schema = []
+
+    if len(prefix_items_schema) < min_items and item_schema is None:
+        raise ValueError(f"Not enough prefixItems ({len(prefix_items_schema)}) to satisfy minItems ({min_items})")
+
+    if max_items is not None and max_items < min_items:
+        raise ValueError(f"maxItems ({max_items}) can't be greater than minItems ({min_items})")
+
+    required_items = []
+    optional_items = []
+
+    n_to_add = max(len(prefix_items_schema), min_items) if max_items is None else max_items
```

Comment:
If I'm reading this correctly, then if `prefix_items` and `max_items` aren't set, then exactly `min_items` will be generated?

Reply:
Ahhh I see the 'infinite tail' below now.

Reply:
Would an earlier comment about this be helpful for clarity?

Reply:
I think it would.